halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/dockerRegistry/DockerRegistryAccountValidator.java

```diff
@@ -1,5 +1,122 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package com.netflix.spinnaker.halyard.config.validate.v1.providers.dockerRegistry;
 
-public class DockerRegistryAccountValidator {
+import com.amazonaws.util.IOUtils;
+import com.netflix.spinnaker.clouddriver.docker.registry.api.v2.client.DockerRegistryCatalog;
+import com.netflix.spinnaker.clouddriver.docker.registry.security.DockerRegistryNamedAccountCredentials;
+import com.netflix.spinnaker.halyard.config.model.v1.node.Validator;
+import com.netflix.spinnaker.halyard.config.model.v1.problem.Problem.Severity;
+import com.netflix.spinnaker.halyard.config.model.v1.problem.ProblemSetBuilder;
+import com.netflix.spinnaker.halyard.config.model.v1.providers.dockerRegistry.DockerRegistryAccount;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DockerRegistryAccountValidator extends Validator<DockerRegistryAccount> {
+  @Override
+  public void validate(ProblemSetBuilder p, DockerRegistryAccount n) {
+    String resolvedPassword = null;
+    String password = n.getPassword();
+    String passwordFile = n.getPasswordFile();
+    String username = n.getUsername();
+
+    boolean passwordProvided = password != null && password.length() > 0;
+    boolean passwordFileProvided = passwordFile != null && passwordFile.length() > 0;
+
+    if (passwordProvided && passwordFileProvided) {
+      p.addProblem(Severity.ERROR, "You have provided both a password and a password file for your docker registry. You can specify at most one");
+      return;
+    }
+
+    try {
+      if (passwordProvided) {
+        resolvedPassword = password;
+      } else if (passwordFileProvided) {
+        resolvedPassword = IOUtils.toString(new FileInputStream(passwordFile));
+
+        if (resolvedPassword.length() == 0) {
+          p.addProblem(Severity.WARNING, "The supplied password file is empty");
+        }
+      } else {
+        resolvedPassword = "";
+      }
+    } catch (FileNotFoundException e) {
+      p.addProblem(Severity.ERROR, "Cannot find provided password file: " + e.getMessage());
+    } catch (IOException e) {
+      p.addProblem(Severity.ERROR, "Error reading provided password file: " + e.getMessage());
+    }
+
+    if (resolvedPassword != null && resolvedPassword.length() > 0) {
+      String message = "Your registry password has %s whitespace, if this is unintentional authentication may fail";
```

Comment:
Might read better as:
"Your registry password has %s whitespace; if this is unintentional, authentication may fail."